File: algorithms/013.roman-to-integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        roman_dict = {"C":100, "M":1000, "D":500, "C":100, "L":50, "X":10, "V":5, "I":1}
        result = 0
        for i in range(0, len(s) - 1):
            prev_char = s[i]
            next_char = s[i+1]
            # 01. next char is bigger than previous one, this is the minus operation
            if (roman_dict[prev_char] < roman_dict[next_char]):
                result -= roman_dict[prev_char]
            # 02. otherwise, it's addition
            else:
                result += roman_dict[prev_char]
        # for the last char, just plus
        result += roman_dict[s[len(s)-1]]
        return result

File: algorithms/013.roman-to-integer.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
	const romanDict = {
		"M": 1000,
		"D": 500,
		"C": 100,
		"L": 50,
		"X": 10,
		"V": 5,
		"I": 1
	};
	
	let result = 0;
	for (let i = 0; i < s.length - 1; i++) {
		const prevChar = s[i];
		const nextChar = s[i + 1];
		// 01. next char is bigger than previous one, this is the minus operation
		if (romanDict[prevChar] < romanDict[nextChar]) {
			result -= romanDict[prevChar];
		}
		// 02. otherwise, it's addition
		else {
			result += romanDict[prevChar];
		}
	}
	// for the last char, just plus
	result += romanDict[s[s.length - 1]];
	return result;
};
